### PR TITLE
feat: introduce Attach Context entrypoint in chat

### PR DESCRIPTION
**Pull request by joyceerhl**

### CODE REVIEW

**Review by justschen — DISMISSED** [review]

**Review by justschen — DISMISSED** [review]

**Review by dbaeumer — APPROVED** [review]

File: src/vs/workbench/contrib/chat/browser/chatInputPart.ts [hunk 1]
```diff
@@ -86,6 +87,16 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
...
 
+	private _onDidChangeAttachedContext = this._register(new Emitter<void>());
+	readonly onDidChangeAttachedContext = this._onDidChangeAttachedContext.event;
+	public get attachedContext() {
+		return this._attachedContext;
+	}
+
+	private readonly _attachedContext = new Set<IChatRequestVariableEntry>();
+
+	private readonly _contextResourceLabels = this.instantiationService.createInstance(ResourceLabels, { onDidChangeVisibility: new Emitter<boolean>().event });
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/chatInputPart.ts:98]: Disposable

File: src/vs/workbench/contrib/chat/browser/chatInputPart.ts [hunk 2]
```diff
@@ -415,16 +423,30 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
...
+					hidePath: true,
+				});
+			} else {
+				label.setLabel(attachment.name);
+			}
 
-	setImplicitContextKinds(kinds: string[]) {
-		dom.setVisibility(this.implicitContextSettingEnabled && kinds.length > 0, this.implicitContextContainer);
-		this.implicitContextLabel.textContent = localize('use', "Use") + ' ' + kinds.map(k => `#${k}`).join(', ');
+			const clearButton = new Button(widget, { supportIcons: true });
```
Comment (roblourens) [inline src/vs/workbench/contrib/chat/browser/chatInputPart.ts:442]: The Button is disposable too

File: src/vs/workbench/contrib/chat/common/chatServiceImpl.ts [hunk 3]
```diff
@@ -436,7 +431,7 @@ export class ChatService extends Disposable implements IChatService {
 
 		this.removeRequest(model.sessionId, request.id);
 
-		await this._sendRequestAsync(model, model.sessionId, request.message, attempt, enableCommandDetection, implicitVariablesEnabled, defaultAgent, location).responseCompletePromise;
+		await this._sendRequestAsync(model, model.sessionId, request.message, attempt, enableCommandDetection, implicitVariablesEnabled, defaultAgent, location, options);
```
[the remark this answers is not in the document]
Comment (joyceerhl) [reply src/vs/workbench/contrib/chat/common/chatServiceImpl.ts:434]: Bad merge, fixing

File: src/vs/workbench/contrib/chat/browser/chatWidget.ts [hunk 4]
```diff
@@ -578,11 +578,11 @@ export class ChatWidget extends Disposable implements IChatWidget {
...
 		}));
-		this._register(this.inputEditor.onDidChangeModelContent(() => this.updateImplicitContextKinds()));
-		this._register(this.chatAgentService.onDidChangeAgents(() => {
-			if (this.viewModel) {
-				this.updateImplicitContextKinds();
+		this._register(this.inputPart.onDidChangeAttachedContext(() => {
+			if (this.bodyDimension) {
+				this.layout(this.bodyDimension.height, this.bodyDimension.width);
 			}
+			this._onDidChangeContentHeight.fire();
```
[the remark this answers is not in the document]
Comment (joyceerhl) [reply src/vs/workbench/contrib/chat/browser/chatWidget.ts:585]: `this.layout` already fires `onDidChangeHeight` so I think firing this event is actually redundant

**Pull request merged**